src/path/fs.rs

```diff
@@ -11,11 +11,14 @@ use std::fs;
 use std::io::{self, Read};
 use std::path;
 
+use std::path::PathBuf;
+use std::path::Path;
+
 use reflection;
 use utils;
 use Predicate;
 
-fn read_file(path: &path::Path) -> io::Result<Vec<u8>> {
+fn read_file<P: AsRef<Path>>(path: P) -> io::Result<Vec<u8>> {
```

Comment:
Only the top-level function should do this.

Its best to minimize what all code lives in generics because the compiler will "monomorphize" it, meaning, it will duplicate the code for every data type used with the generics.  

See http://www.suspectsemantics.com/blog/2016/12/03/monomorphization-bloat/